setools/policyrep/terule.pxi

```diff
@@ -142,6 +142,26 @@ cdef class AVRule(BaseTERule):
         """The rule's default type."""
         raise RuleUseError("{0} rules do not have a default type.".format(self.ruletype))
 
+    def create_expanded(self, source, target, perms):
```

Comment:
I'd prefer this to be named `derive_expanded`. It should do thorough checking to ensure it is a true expansion of the rule:
1. source and target are proper policyrep objects (can add type to method signature, e.g. `derive_expanded(self, BaseType source, ...)`
2. the source/target parameter `is` (not `==`) self.source/target or the parameter is `in` self.source/target. I added a `__contains__` method in master to `TypeAttribute` which should make this doable.
3. Rule is not expanded (don't return `self`)

If 2 or 3 fail, then it is a `RuleUseError`. The compiler will add error code for 1.

Why is a perms parameter needed? If perms != self.perms, it is not a true expansion of the origin rule.

Reply:
The permissions are usually the same, but if only some of the permissions of a rule are redundant, then the rule remains, but with the redundant permissions removed.

Reply:
I am now checking if perms is a subset of self.perms and returning RuleUseError if it is not.
I fixed the other things mentioned here as well.